### DISCUSSION TITLE
Is there support for multiple Dbs per app?

### DISCUSSION THREAD

**Discussion by AFDevMike** [cyrilletuzi/angular-async-local-storage#98] [General]
Is there support for multiple Dbs per app?

Following: https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/PREFIX.md I created 2 modules each providing a different prefix _in the same app_ but the data does not store segregated as I expected.

We are on TS 3.2+ and so can't update to Version 7 and am looking for a mechanism to delete `.clear()` groups of related entities. 

Thank you for this excellent product!

**Comment by cyrilletuzi** [comment]
Hi @AFDevMike, thanks for your interest in this lib.

Currently, v7.4+ of the lib is needed. The recipe for multiple storages is [here](https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/MAP_OPERATIONS.md#multiple-stores).

Note that v7 is not recommended, but it can work with TypeScript 3.2 if you work around the TypeScript issue (see #64).

v8 (tracked in #76) should be fully compatible with Angular 7 (and TS 3.1 & 3.2). Note that the behavior of `keys()` may change in v8 (currently it is returning all the keys *at once*, maybe v8 will *iterate* on keys, I still need to investigate this).

A real feature for multiple databases would be a good idea,  but:

1. it needs some design: I would suggest to stay as close as possible to the upcoming kv-storage standard, which has a [multiple databases feature](https://github.com/WICG/kv-storage#new-storagearea-separate-storage-areas). But it should be adapted to Angular. As a service in Angular is a singleton, I'm not sure what would be the best practice to handle multiple storage services.

2. I currently have not much time for new features (upcoming v8 is already a lot of work). So 3 possibilities:
- wait, with no guarantees,
- do a PR (after having agreed on API design),
- see if your company would be OK to [sponsor](https://github.com/sponsors/cyrilletuzi) this so I can include it in my work time.

**Comment by cyrilletuzi** [comment]
For reference: this request was already made and there was already some investigation in #39 with @juristr and  @dphochman.

v7 already answered to this use case, but the upcoming kv-storage opened new options of API design.

**Comment by cyrilletuzi** [comment]
For documentation: multiple databases can already be simulated, see [the recipe here](https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/MAP_OPERATIONS.md).

v8 will ease this recipe.
